pkg/api/controller.go

```diff
@@ -556,6 +556,40 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body apigen.L
 	writeResponse(w, r, http.StatusOK, response)
 }
 
+func (c *Controller) ExternalPrincipalLogin(w http.ResponseWriter, r *http.Request, body apigen.ExternalPrincipalLoginJSONRequestBody) {
+	ctx := r.Context()
+	if c.isExternalPrincipalNotSupported(ctx) {
+		writeError(w, r, http.StatusNotImplemented, "Not implemented")
+		return
+	}
+
+	c.LogAction(ctx, "external_principal_login", r, "", "", "")
+	principalID, err := c.Authentication.ExternalPrincipalLogin(ctx, body)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	user, err := c.Auth.GetUserByExternalID(ctx, principalID)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+
+	loginTime := time.Now()
+	duration := c.Config.Auth.LoginDuration
+	expires := loginTime.Add(duration)
+	secret := c.Auth.SecretStore().SharedSecret()
+
+	tokenString, err := GenerateJWTLogin(secret, *user.FriendlyName, loginTime.Unix(), expires.Unix())
```

Comment:
expiration should have: 
- configuration: max value (i.e 24 hours) 
- configuration: default value (i.e 12 hours) 
- optional: per login request value